server/src/main/java/org/opensearch/search/suggest/completion/CompletionSuggester.java

```diff
@@ -108,15 +109,21 @@ private static void suggest(IndexSearcher searcher, CompletionQuery query, TopSu
         for (LeafReaderContext context : searcher.getIndexReader().leaves()) {
             BulkScorer scorer = weight.bulkScorer(context);
             if (scorer != null) {
+                LeafCollector leafCollector = null;
```

Comment:
I just realized this fix is simpler.

Per this [Query#rewrite deprecation](https://github.com/apache/lucene/blob/branch_9x/lucene/core/src/java/org/apache/lucene/search/Query.java#L98) just change line 106 to the following

```suggestion
                    query = (CompletionQuery) query.rewrite(searcher);
```

You're basically reimplementing here the finish hook that was added in [Lucene PR#12380](https://github.com/apache/lucene/pull/12380). 

That logic path wasn't being executed though because we're using the old collector API through the creation of a new searcher wrapping the reader.

I forgot about this deprecation in the 9.7 upgrade and it wasn't exposed because #12380 wasn't merged until after 9.7.

Reply:
Sorry about it, merged just before, there is another issue I run into, again with Suggester but slightly different cause:

```
java.lang.IllegalArgumentException: onlrtdbizi is not a SuggestField
	at org.apache.lucene.search.suggest.document.CompletionWeight.bulkScorer(CompletionWeight.java:80)
	at org.opensearch.search.suggest.completion.CompletionSuggester.suggest(CompletionSuggester.java:110)
	at org.opensearch.search.suggest.completion.CompletionSuggester.innerExecute(CompletionSuggester.java:78)
	at org.opensearch.search.suggest.completion.CompletionSuggester.innerExecute(CompletionSuggester.java:1)
```

I could incorporate your suggestion into the upcoming fix (still trying to figure out what is happening)

Reply:
I'm finishing a PR that cuts over all rewrites to the non-deprecated one. It includes this change in the CompletionSuggester and other places.

Reply:
I opened [the PR](https://github.com/opensearch-project/OpenSearch/pull/8967) but I left your patch in. I was wrong, there's still an issue and cutting over to invoke the hook doesn't solve the occasional failure. Let me know if that's what you're investigating.

Reply:
> I opened [the PR](https://github.com/opensearch-project/OpenSearch/pull/8967) but I left your patch in. I was wrong, there's still an issue and cutting over to invoke the hook doesn't solve the occasional failure.

Do you have a sample stack trace or logs by any chance?

> Let me know if that's what you're investigating.

The one that caused another batch of flakyness: https://github.com/opensearch-project/OpenSearch/pull/8968